packages/kit/src/runtime/client/client.js

```diff
@@ -1241,30 +1237,54 @@ function get_rerouted_url(url) {
  * @param {URL | undefined} url
  * @param {boolean} invalidating
  */
-function get_navigation_intent(url, invalidating) {
+async function get_navigation_intent(url, invalidating) {
 	if (!url) return;
 	if (is_external_url(url, base, app.hash)) return;
 
-	const rerouted = get_rerouted_url(url);
-	if (!rerouted) return;
-
-	const path = get_url_path(rerouted);
+	if (__SVELTEKIT_CLIENT_ROUTING__) {
+		const rerouted = get_rerouted_url(url);
+		if (!rerouted) return;
+
+		const path = get_url_path(rerouted);
+
+		for (const route of routes) {
+			const params = route.exec(path);
+
+			if (params) {
+				const id = get_page_key(url);
+				/** @type {import('./types.js').NavigationIntent} */
+				const intent = {
+					id,
+					invalidating,
+					route,
+					params: decode_params(params),
+					url
+				};
+				return intent;
```

Comment:
tangential to this PR but could we just use an explicit `@returns` annotation instead of this ceremony?

```suggestion
				return {
					id,
					invalidating,
					route,
					params: decode_params(params),
					url
				};
```